### PR TITLE
feature/BIT-579/Adding Prometheus

### PR DESCRIPTION
**Pull request by eduardogr**
## What are you going to find in this PR?

Basically this PR is the code develop by @unconst in the previous PR: https://github.com/opentensor/bittensor/pull/894.

I have just moved the changes into a brand new branch to focus on prometheus changes. You are going to find same code as there regarding prometheus and none regarding other changes, i've omitted non-prometheus changes.

## What are you not going to find in this PR?

- wandb removal

### CODE REVIEW

File: bittensor/_axon/__init__.py [hunk 1]
```diff
@@ -35,90 +35,92 @@
 from . import axon_impl
 
 class axon:
-    """ The factor class for bittensor.Axon object
-    The Axon acts a grpc server for the bittensor network and allows for communication between neurons.
-    By default, the grpc server follows the bittensor protocol and transports forward and backwards requests
-    between validators and servers. 
+    """ The factory class for bittensor.Axon object
+    The Axon is a grpc server for the bittensor network which opens up communication between it and other neurons.
+    The server protocol is defined int bittensor.proto and describes the manner in which forward and backwards requests
```
Comment (unconst) [inline bittensor/_axon/__init__.py]: this is my mistake, but ive added a typo here " is defined in" not " is defined int "

**Review by unconst — APPROVED** [review]
LFG !!!! Stoked for this

**Review by Eugene-hu — APPROVED** [review]
LGTM

**Review by opentaco — APPROVED** [review]

File: bittensor/_axon/__init__.py (hunk 1 above)
Comment (eduardogr) [reply bittensor/_axon/__init__.py]: Thanks ! solved it

**Pull request merged**